### DISCUSSION TITLE
ZeroMatchesError for DRAGEN haplotyper generated WGS gVCF

### DISCUSSION THREAD

**Discussion by kpalanikannan** [PGScatalog/pgsc_calc#369] [Q&A]
Hi Team,

I am using the WGS gVCF created by DRAGEN haplotypecaller. I am supplying seperate gVCF for each chromosome through samplesheet.  I provided genotype column as GT and file type as VCF. I use GRCh38 build gVCF. My catalog is converted to GRCh38 build including header information (PGS001804_GRCh38.txt) and column names as expected. I derived this from "PGS001804_hmPOS_GRCh38.txt". I defined the --target as GRCh38. Provided example lines in my gVCF below for hom-ref and het-alt variants. For continuous stretch of hom-ref scenario, DRAGEN creates hom-ref blocks.

```
10      10901   .       T       <NON_REF>       .       PASS    END=10901       GT:AD:DP:GQ:MIN_DP:PL:SPL:ICNT  0/0:20,2:22:35:22:0,35,730:0,35,255:19,0
10      10902   .       T       <NON_REF>       .       PASS    END=10903       GT:AD:DP:GQ:MIN_DP:PL:SPL:ICNT  0/0:21,1:22:50:22:0,50,696:0,50,255:20,0
10      10904   .       C       G,<NON_REF>     12.75   PASS    DP=20;MQ=21.8;MQRankSum=2.431;ReadPosRankSum=1.823;FractionInformativeReads=1   GT:AD:AF:DP:F1R2:F2R1:GQ:PL:SPL:ICNT:GOP:GP:PRI:SB:MB   0/1:9,11,0:0.55,0:20:3
,10,0:6,1,0:12:47,0,19,426,333,759:255,0,255:8,0:40,40:12.746,0.2677,21.777,426.34,367.56,762.12:0,34.77,37.77,34.77,69.54,37.77:7,2,10,1:5,4,9,2
```
- [x] Scores file created
- [x] PLINK2VCF processes was successful and pgen files are created.
- [x] I see that "*.pvar.zst" files are created for each chromosome from MATCH_VARIANTS process. 
- [ ] But *.ipc.zst files are generated with 8kb in size without any error message from MATCH_VARIANTS in ".command.err" or ".command.log"
- [ ] In the MATCH_COMBINE step, it fails with the error `pgscatalog.core.lib.pgsexceptions.ZeroMatchesError: No match candidates found for any scoring files`.

Can you help to find the issue and fix? How do I see the failed matches?

**Comment by smlmbrt** [comment]
@kpalanikannan, you might find the discussion in #123 and https://github.com/PGScatalog/pgsc_calc/discussions/247 informative, basically the currently matching pipeline can't handle missing non-reference alleles. In your case only the 4th row would be eligible for matching with a C/G. It's a long-term goal of the pipeline to add this functionality (#247 proposes a workaround).
